src/compiler/filesystem.cc

```diff
@@ -141,13 +141,14 @@ void Filesystem::canonicalize(char* path) {
       path[canonical_pos++] = path[i++];
     }
   }
-  // Drop trailing path seperator.
-  // There can only be one.
-  if (path[canonical_pos - 1] == path_separator()) {
+  // Drop trailing path seperator unless it's the root.
+  path[canonical_pos] = '\0';
+  if (!(is_absolute && is_root(path)) && path[canonical_pos - 1] == path_separator()) {
```

Comment:
This confuses me. Can `is_absolute` be false and still allow `is_root` to return true?

Reply:
No. it's just to avoid to do the is_root check when we already know that the path isn't even absolute.
I removed the check.
done.